frontend/packages/console-metal3/src/plugin.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  Plugin,
+  ResourceNSNavItem,
+  ResourceListPage,
+  ResourceDetailPage,
+  ModelBasedFeatureFlag
+} from '@console/plugin-sdk';
+
+import { BaremetalHostModel } from './models';
+
+type ConsumedExtensions =
+  | ResourceNSNavItem
+  | ResourceListPage
+  | ResourceDetailPage
+  | ModelBasedFeatureFlag;
+
+const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'FeatureFlag/ModelBased',
+    properties: {
+      model: BaremetalHostModel,
+      flag: 'METAL3'
+    }
+  },
+  {
+    type: 'NavItem/ResourceNS',
+    properties: {
+      section: 'Compute',
+      componentProps: {
+        name: 'Bare Metal Hosts',
+        resource: 'baremetalhosts'
+      }
+    }
+  },
+  {
+    type: 'ResourcePage/List',
+    properties: {
+      model: BaremetalHostModel,
+      loader: () =>
+        import(
+          './components/host' /* webpackChunkName: "baremetalhost" */
```

Comment:
The `webpackChunkName` value should reflect the name of your plugin for better traceability.

I'd suggest to start with a single split point, i.e. use `webpackChunkName: "metal3"` for all async module requests.

Reply:
We typically use different chunk names for different pages for better code splitting. Then the JS is only loaded when you go to that page.

Reply:
OK, sounds reasonable :smiley:

Looking at `public/components/resource-pages.ts` the convention seems to be to put both resource list and detail pages under the same split point, e.g. `pod` for both `PodsPage` and `PodsDetailsPage` (group by resource).

Reply:
I've used `metal3-baremetalhost`